src/proposals/Addresses.sol

```diff
@@ -335,6 +335,10 @@ contract Addresses is IAddresses, Test {
                 )
             )
         );
+
+        if (chainId == block.chainid && data.isContract) {
```

Comment:
I think these checks are redundant as these are already there in the `_checkAddress`. So we can remove this if block.